pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -180,27 +184,34 @@ func (d *DualWriterMode1) Update(ctx context.Context, name string, objInfo rest.
 
 	go func() {
 		ctx, cancel := context.WithTimeoutCause(ctx, time.Second*10, errors.New("storage update timeout"))
```

Comment:
`defer cancel()` should be right after this line, and we should remove the `defer cancel` below.